examples/noise-js/index.js

```diff
@@ -0,0 +1,357 @@
+      });
+      const payload = ProtoChatMessage.encode(message).finish();
+
+      await node.lightPush.push(encoder, { payload, timestamp });
+    });
+  } catch (err) {
+    ui.waku.error(err.message);
+    ui.hide();
+  }
+}
+
+function buildPairingURLFromObj(pairingObj) {
+  const pInfo = pairingObj.getPairingInfo();
+
+  // Data to encode in the QR code. The qrMessageNametag too to the QR string (separated by )
+  const messageNameTagParam = `messageNameTag=${utils.bytesToHex(
+    pInfo.qrMessageNameTag
+  )}`;
+  const qrCodeParam = `qrCode=${encodeURIComponent(pInfo.qrCode)}`;
+
+  return `${window.location.href}?${messageNameTagParam}&${qrCodeParam}`;
+}
+
+function getPairingInfoFromURL() {
+  const urlParams = new URLSearchParams(window.location.search);
+
+  const messageNameTag = urlParams.get("messageNameTag");
+  const qrCodeString = urlParams.get("qrCode");
+
+  if (!(messageNameTag && qrCodeString)) {
+    return undefined;
+  }
+
+  return new noise.InitiatorParameters(
+    decodeURIComponent(qrCodeString),
+    utils.hexToBytes(messageNameTag)
+  );
+}
+
+function getSenderAndResponder(node) {
+  const sender = {
+    async publish(encoder, msg) {
+      await node.lightPush.push(encoder, msg);
+    },
+  };
+
+  const msgQueue = new Array();
+  const subscriptions = new Map();
+  const intervals = new Map();
+
+  const responder = {
+    async subscribe(decoder) {
+      const subscription = await node.filter.subscribe(
+        [decoder],
+        (wakuMessage) => {
+          msgQueue.push(wakuMessage);
+        }
+      );
+      subscriptions.set(decoder.contentTopic, subscription);
+    },
+    async nextMessage(contentTopic) {
+      if (msgQueue.length != 0) {
+        const oldestMsg = msgQueue.shift();
+        if (oldestMsg.contentTopic === contentTopic) {
+          return oldestMsg;
+        }
+      }
+
+      return new Promise((resolve) => {
+        const interval = setInterval(() => {
+          if (msgQueue.length != 0) {
+            clearInterval(interval);
+            const oldestMsg = msgQueue.shift();
+            if (oldestMsg.contentTopic === contentTopic) {
+              resolve(oldestMsg);
+            }
+          }
+        }, 100);
+        intervals.set(contentTopic, interval);
+      });
+    },
+    async stop(contentTopic) {
+      if (intervals.has(contentTopic)) {
+        clearInterval(intervals.get(contentTopic));
+        intervals.delete(contentTopic);
+      }
+      if (subscriptions.has(contentTopic)) {
+        await subscriptions.get(contentTopic)();
+        subscriptions.delete(contentTopic);
+      } else {
+        console.log("Subscriptipon doesnt exist");
+      }
+    },
+  };
+
+  return [sender, responder];
+}
+
+async function scheduleHandshakeAuthConfirmation(pairingObj, ui) {
+  const authCode = await pairingObj.getAuthCode();
+  ui.handshake.connecting();
+  pairingObj.validateAuthCode(confirm("Confirm that authcode is: " + authCode));
```

Comment:
nitpick: Buttons displayed in the pop up are "Ok" and "Cancel" might be more appropriate to have: "Confirm" and "Cancel".